### ISSUE TITLE
protobuf/gcc errors on windows

### ISSUE THREAD

**Issue by lrettig** [spacemeshos/go-spacemesh#1954]
## Description
Running into the below error when trying to run `make install` on Windows

## Steps To Reproduce
1. Install prereqs
1. Clone repository
1. Run `make install`

## Environment
 - OS: Windows 10
 - Node Version: building from head eaea050e3e3f4404481aa9b231f4e7e14af67cef

## Additional Resources 
```
PS > make install
process_begin: CreateProcess(NULL, bash -c "if [ \"$TRAVIS_PULL_REQUEST\" == \"false\" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi", ...) failed.
setup_env.bat
'protoc' is not recognized as an internal or external command,
operable program or batch file.
Fetching protoc...
Extracting protoc...

SUCCESS: Specified value was saved.
installing protoc-gen-go...
can't load package: package /protoc-gen-go is not in GOROOT (c:\go\src\protoc-gen-go)
installing protoc-gen-grpc-gateway...
can't load package: package /protoc-gen-grpc-gateway is not in GOROOT (c:\go\src\protoc-gen-grpc-gateway)
installing protoc-gen-swagger...
can't load package: package /protoc-gen-swagger is not in GOROOT (c:\go\src\protoc-gen-swagger)
make: *** [install] Error 1
```

**Comment by lrettig** [comment]
This seems to work in powershell, I'm not sure why `setup_env.bat` isn't working:

```
> go list -m -f github.com/golang/protobuf
github.com/golang/protobuf
```

**Comment by lrettig** [comment]
Opened #2028 to track the fixes I'm making (it's based on #1955). I fixed the error described above (e0f28310cb87dce75cc662653e4edec9ae41aa02) but I'm getting another error now:

```
> .\scripts\win\genproto.bat
Could not find protoc-gen-go. If you've run `make install`, ensure that $GOPATH\bin is in your shell's $PATH.
```

This is coming from `scripts/win/verify-protoc-gen-go.ps1` and is happening despite the fact that go was installed correctly and the path is configured correctly.

**Comment by lrettig** [comment]
Fixed. Running into yet another error:

```
> .\scripts\win\genproto.bat
Generating protobuf for api/pb
C:\Users\lanerettig\go\src\github.com\grpc-ecosystem\grpc-gateway\third_party\googleapis: warning: directory does not exist.
google/api/annotations.proto: File not found.
api.proto: Import "google/api/annotations.proto" was not found or had errors.
C:\Users\lanerettig\go\src\github.com\grpc-ecosystem\grpc-gateway\third_party\googleapis: warning: directory does not exist.
google/api/annotations.proto: File not found.
api.proto: Import "google/api/annotations.proto" was not found or had errors.
C:\Users\lanerettig\go\src\github.com\grpc-ecosystem\grpc-gateway\third_party\googleapis: warning: directory does not exist.
google/api/annotations.proto: File not found.
api.proto: Import "google/api/annotations.proto" was not found or had errors.

> dir C:\Users\lanerettig\go


    Directory: C:\Users\lanerettig\go


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        5/21/2020   3:57 PM                bin
d-----        5/21/2020   3:42 PM                pkg
```

**Comment by lrettig** [comment]
Fixed this (8273478d2ad2702b917ceff3ab80b437f7870be4).

Running into another error now that doesn't appear to be protobuf-related:

```
> make build
process_begin: CreateProcess(NULL, bash -c "if [ \"$TRAVIS_PULL_REQUEST\" == \"false\" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi", ...) failed.
scripts\win\genproto.bat
Generating protobuf for api/pb
process_begin: CreateProcess(NULL, cat version.txt, ...) failed.
go build -ldflags "-X main.version= -X main.commit=8273478d2ad2702b917ceff3ab80b437f7870be4 -X main.branch=fix-windows-setup" -o Z:\Documents\Github\go-spacemesh/build/go-spacemesh.exe
# github.com/spacemeshos/go-spacemesh/cmd/node
exec: "gcc": executable file not found in %PATH%
make: *** [build] Error 2
```

https://stackoverflow.com/questions/43580131/exec-gcc-executable-file-not-found-in-path-when-trying-go-build suggests installing https://jmeubank.github.io/tdm-gcc/.
